jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -595,7 +595,7 @@ protected boolean passConditionalHeaders(HttpServletRequest request, HttpServlet
                     return false;
                 }
 
-                long ifmsl = request.getDateHeader(HttpHeader.IF_MODIFIED_SINCE.asString());
+                long ifmsl = DateParser.parseDate(ifms);
                 if (ifmsl != -1 && content.getResource().lastModified() / 1000 <= ifmsl / 1000)
```

Comment:
Per https://www.rfc-editor.org/rfc/rfc9110#section-13.1.3-7

> A recipient MUST ignore the If-Modified-Since header field if the resource does not have a modification date available.

The value of `content.getResource().lastModified()` should be checked to know if the resource has a modification date and skip this check if it has no modification date.

Reply:
done